### PR TITLE
Fixes #32611: Drop Puppet 5 support

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ehelms** [comment]
Testing update needed prior to -- https://github.com/theforeman/jenkins-jobs/pull/65

File: modules/kafo_configure/manifests/init.pp [hunk 1]
```diff
@@ -4,14 +4,7 @@
 #   $modulepath/config/answers.yaml
 #   /etc/kafo-configure/answers.yaml
 #
-# @param add_progress
-#   Whether to add a progress bar. Only works on Puppet < 6.
 class kafo_configure(
-  Boolean $add_progress = $::kafo_add_progress,
```
Comment (ekohl) [inline modules/kafo_configure/manifests/init.pp:10]: If you're dropping this, please also drop all code for it:
https://github.com/theforeman/kafo/blob/a45b1ebedda66cdf460c3fe5bfbb3696f2bd198f/lib/kafo/puppet_command.rb#L76
https://github.com/theforeman/kafo/blob/a45b1ebedda66cdf460c3fe5bfbb3696f2bd198f/lib/kafo/puppet_command.rb#L82-L84
(and the tests)

**Review by ekohl — CHANGES_REQUESTED** [review]
Please keep the README up to date (https://github.com/theforeman/kafo#requirements).

**Comment by ekohl** [comment]
Also, I'd like to hold this off. Doing another major release is just a pain and we don't really gain a lot by doing so now, right?

**Comment by ehelms** [comment]
> Also, I'd like to hold this off. Doing another major release is just a pain and we don't really gain a lot by doing so now, right?

If not now, then when? If we plan to drop this support, we benefit ourselves from doing it across the board and at the start of a release. The "major" bump aspect of it is pretty easy to orchestrate. What do you see as the pain ?

**Comment by ekohl** [comment]
Cherry picks become more work.

**Comment by ekohl** [comment]
In https://github.com/theforeman/kafo/pull/354 I went a step further and also cleaned up the progress bar support.

**Pull request closed without merging**